### PR TITLE
Fix: fix tests due to sqlglot upgrade

### PR DESCRIPTION
**Pull request by tobymao**

### CODE REVIEW

File: sqlmesh/core/renderer.py [hunk 1]
```diff
@@ -417,7 +417,13 @@ def _optimize_query(self, query: exp.Subqueryable) -> exp.Subqueryable:
...
                             "",
                         ):
-                            select.replace(exp.alias_(select, select.output_name))
+                            alias = exp.alias_(select, select.output_name)
+                            comments = alias.this.comments
+                            if comments:
+                                alias.add_comments(comments)
+                                comments.clear()
+
+                            select.replace(alias)
```
Comment (GeorgeSittas) [inline sqlmesh/core/renderer.py:426]: Not sure why these changes are needed. What was the issue? I put a breakpoint in the failing test and I couldn't spot a difference in the ASTs so it's weird the comment was generated before the comma.

Comment (GeorgeSittas) [reply sqlmesh/core/renderer.py:426]: how did the test work before though?

Comment (tobymao) [reply sqlmesh/core/renderer.py:426]: because it compares the ast only

Comment (GeorgeSittas) [reply sqlmesh/core/renderer.py:426]: Good catch

**Pull request merged**